transformations/emoji_icon_transformation/transformation.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+import random
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+def convert(perturbed_text, dict1, dict2):
+    for k in dict1:
+        for s in dict1[k]:
+            if s in perturbed_text:
+                perturbed_text = perturbed_text.replace(
+                    s, random.choice(dict2[k])
+                )
+    return perturbed_text
+
+
+def emoji2icon(
+    text, text2emoji, text2icon, seed=42, max_outputs=1, emoji_to_icon=True
+):
+    random.seed(seed)
+
+    perturbed_texts = []
+    for _ in range(max_outputs):
+        perturbed_text = text
+        if emoji_to_icon:
+            perturbed_text = convert(perturbed_text, text2emoji, text2icon)
+        else:
+            perturbed_text = convert(perturbed_text, text2icon, text2emoji)
+        perturbed_texts.append(perturbed_text)
+    return perturbed_texts
+
+
+class EmojiToIcon(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["All"]
+    keywords = ["rule-based", "visual", "high-precision", "low-coverage"]
+
+    def __init__(self, seed=42, max_outputs=1, emoji_to_icon=False):
```

Comment:
Perhaps add some type hinting and comments here? Just a small detail but always makes the code better :)

Reply:
Thank you @Filco306 for the suggestion, I have added type hinting and some small comments in 5f610c740a9f32aeaa20c59952e2560240a61dcb. Now I agree that the code is much easier to understand!